File: backend/BoundfoxStudios.Smudgy.Data/Entities/BaseEntity.cs
using System;

namespace BoundfoxStudios.Smudgy.Data.Entities
{
  public abstract class BaseEntity
  {
    private int ClusterId { get; set; }
    public Guid Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
  }
}

File: backend/src/modules/database/entities/base.entity.ts
import { EntitySchemaOptions } from 'typeorm';
import { Guid } from '../../../models/guid';

export interface BaseEntity {
  clusterId: number;
  id: Guid;
  createdAt: Date;
  updatedAt: Date;
}

const baseEntitySchema: EntitySchemaOptions<BaseEntity> = {
  name: 'base-entity',
  columns: {
    id: {
      type: String,
      unique: true,
    },
    clusterId: {
      type: Number,
      primary: true,
      generated: true,
    },
    createdAt: {
      type: Date,
      createDate: true,
    },
    updatedAt: {
      type: Date,
      updateDate: true,
    },
  },
};

export const BaseEntityColumnScheme = baseEntitySchema.columns;
